reporter/amqp/amqp.go

```diff
@@ -0,0 +1,201 @@
+/*
+Package amqp implements a RabbitMq reporter to send spans to a Rabbit server/cluster.
+*/
+package amqp
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+
+	"github.com/streadway/amqp"
+
+	"github.com/openzipkin/zipkin-go/model"
+	"github.com/openzipkin/zipkin-go/reporter"
+)
+
+// defaultRmqRoutingKey/Exchange/Kind sets the standard RabbitMQ queue our Reporter will publish on.
+const (
+	defaultRmqRoutingKey = "zipkin"
+	defaultRmqExchange   = "zipkin"
+	defaultExchangeKind  = "direct"
+)
+
+// rmqReporter implements Reporter by publishing spans to a RabbitMQ exchange
+type rmqReporter struct {
+	e        chan error
+	channel  *amqp.Channel
+	conn     *amqp.Connection
+	Exchange string
+	Queue    string
+	logger   *log.Logger
+}
+
+// ReporterOption sets a parameter for the rmqReporter
+type ReporterOption func(c *rmqReporter)
+
+// Logger sets the logger used to report errors in the collection
+// process.
+func Logger(logger *log.Logger) ReporterOption {
+	return func(c *rmqReporter) {
+		c.logger = logger
+	}
+}
+
+// Exchange sets the Exchange used to send messages (
+// see https://github.com/openzipkin/zipkin/tree/master/zipkin-collector/rabbitmq
+// if want to change default routing key or exchange
+func Exchange(e string) ReporterOption {
+	return func(c *rmqReporter) {
+		c.Exchange = e
+	}
+}
+
+// Queue sets the Queue used to send messages
+func Queue(t string) ReporterOption {
```

Comment:
Is this `t` name for the variable right?

Reply:
> Is this `t` name for the variable right?

Yep, will rename to queue